pandapipes/component_models/abstract_models/circulation_pump.py

```diff
@@ -43,30 +113,34 @@ def extract_results(cls, net, options, branch_results, nodes_connected, branches
         :type options:
         :return: No Output.
         """
-        res_table, circ_pump, index_nodes_from, node_pit, _ = \
-            super().extract_results(net, options, None, nodes_connected, branches_connected)
+        circ_pump_tbl = net[cls.table_name()]
 
-        index_juncts_to = circ_pump.to_junction.values
-        junct_uni_to = np.array(list(set(index_juncts_to)))
-        index_nodes_to = get_lookup(net, "node", "index")[
-            cls.get_connected_node_type().table_name()][junct_uni_to]
+        if len(circ_pump_tbl) == 0:
+            return
 
-        deltap_bar = node_pit[index_nodes_from, PINIT] - node_pit[index_nodes_to, PINIT]
-        res_table["deltap_bar"].values[:] = deltap_bar
+        res_table = net["res_" + cls.table_name()]
 
-    @classmethod
-    def get_connected_junction(cls, net):
-        junction = net[cls.table_name()].from_junction
-        return junction
+        branch_pit = net['_pit']['branch']
+        node_pit = net["_pit"]["node"]
 
-    @classmethod
-    def get_result_table(cls, net):
-        """
+        junction_lookup = get_lookup(net, "node", "index")[
+            cls.get_connected_node_type().table_name()]
+        fn_col, tn_col = cls.from_to_node_cols()
+        # get indices in internal structure for flow_junctions in circ_pump tables which are
+        # "active"
+        flow_junctions = circ_pump_tbl[tn_col].values
+        flow_nodes = junction_lookup[flow_junctions]
+        in_service = circ_pump_tbl.in_service.values
+        p_grids = np.isin(circ_pump_tbl.type.values, ["p", "pt"]) & in_service
+        sum_mass_flows, inverse_nodes, counts = get_mass_flow_at_nodes(net, node_pit, branch_pit,
+                                                                       flow_nodes[p_grids], cls)
 
-        :param net: The pandapipes network
-        :type net: pandapipesNet
-        :return: (columns, all_float) - the column names and whether they are all float type. Only
-                if False, returns columns as tuples also specifying the dtypes
-        :rtype: (list, bool)
-        """
-        return ["mdot_kg_per_s", "deltap_bar"], True
+        # positive results mean that the circ_pump feeds in, negative means that the ext grid
+        # extracts (like a load)
+        res_table["mdot_flow_kg_per_s"].values[p_grids] = - (sum_mass_flows / counts)[inverse_nodes]
+
+        return_junctions = circ_pump_tbl[fn_col].values
+        return_nodes = junction_lookup[return_junctions]
+
+        deltap_bar = node_pit[flow_nodes, PINIT] - node_pit[return_nodes, PINIT]
+        res_table["deltap_bar"].values[in_service] = deltap_bar[in_service]
```

Comment:
This is again due to how the diff is displayed. I also reorganized the structure of the circulation_pump class, so that the methods appear in more or less the same order in all components.